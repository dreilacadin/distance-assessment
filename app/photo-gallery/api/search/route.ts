import { NextRequest, NextResponse } from "next/server";

/** Gets only 1 photo object using id */
export async function GET(request: NextRequest) {
  const searchParams = request.nextUrl.searchParams;
  const id = searchParams.get("id");
  let data = await fetch(`https://api.unsplash.com/photos/${id}`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Client-ID ${process.env.UNSPLASH_ACCESS_KEY}`,
    },
  });

  let image = await data.json();

  return NextResponse.json(image);
}
