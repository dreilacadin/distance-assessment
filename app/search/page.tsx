import SearchBar from "@/app/components/inputs/SearchBar";
import SearchResults from "@/app/search/_components/SearchResults";

/**
 * Search Page which is handled by passing a search query using URL params
 *
 * Why URL Params?
 * 1. Bookmarkable and Shareable URs: Since the search parameters are in the URL, users can bookmark the current state of the application, including their search queries and filters, for future reference or sharing.
 * 2. Server-Side Rendering and Initial Load: URL parameters can be directly consumed on the server to render the initial state, making it easier to handle server rendering.
 * 3. Analytics and Tracking: Having search queries and filters directly in the URL makes it easier to track user behavior without requiring additional client-side logic.
 *  */
export default async function SearchPage({
  searchParams,
}: {
  searchParams: Promise<{ query: string }>;
}) {
  const { query } = await searchParams;
  return (
    <>
      <SearchBar />
      <SearchResults query={query} />
    </>
  );
}
