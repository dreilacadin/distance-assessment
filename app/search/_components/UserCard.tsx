import { User } from "@/app/search/dummyUsers";
import {
  BriefcaseIcon,
  EnvelopeIcon,
  GlobeAltIcon,
  MapPinIcon,
  PhoneIcon,
  UserCircleIcon,
} from "@heroicons/react/24/outline";

export default function UserCard({ user }: { user: User }) {
  return (
    <div className="my-8 rounded-lg border border-gray-200 bg-white p-8 shadow">
      <ul className="flex flex-col space-y-2 text-slate-800">
        <li className="flex items-center space-x-2">
          <UserCircleIcon className="size-5 text-gray-700" />
          <span className="text-lg font-semibold">{user.name}</span>
        </li>
        <li className="flex items-center space-x-2">
          <EnvelopeIcon className="size-5 text-gray-700" />
          <span data-testid="email">{user.email}</span>
        </li>
        <li className="flex items-center space-x-2">
          <MapPinIcon className="size-5 text-gray-700" />
          <div>
            <span>{user.address.street}, </span>
            <span>{user.address.suite}, </span>
            <span>{user.address.city}, </span>
            <span>{user.address.zipcode}</span>
          </div>
        </li>
        <li className="flex items-center space-x-2">
          <PhoneIcon className="size-5 text-gray-700" />
          <span>{user.phone}</span>
        </li>
        <li className="flex items-center space-x-2">
          <GlobeAltIcon className="size-5 text-gray-700" />
          <span>{user.website}</span>
        </li>
        <li className="flex items-center space-x-2">
          <BriefcaseIcon className="size-5 text-gray-700" />
          <p>{user.company.name}</p>
          <p>{user.company.catchPhrase}</p>
        </li>
      </ul>
    </div>
  );
}
