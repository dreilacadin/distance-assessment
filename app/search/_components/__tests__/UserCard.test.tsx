import UserCard from "@/app/search/_components/UserCard";
import { users } from "@/app/search/dummyUsers";
import { render, screen } from "@testing-library/react";

describe("UserCard unit tests", () => {
  it("should render a user card when passed a user object", () => {
    const { container } = render(<UserCard user={users[0]} />);

    /** Should render the correct user email */
    expect(screen.getByTestId("email")).toHaveTextContent(users[0].email);
    expect(container).toMatchSnapshot();
  });

  it("should render another user card correctly when passed another user object", () => {
    const { container } = render(<UserCard user={users[8]} />);

    /** Should render the correct user email */
    expect(screen.getByTestId("email")).toHaveTextContent(users[8].email);
    expect(container).toMatchSnapshot();
  });
});
