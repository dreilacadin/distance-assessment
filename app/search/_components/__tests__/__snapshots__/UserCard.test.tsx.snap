// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard unit tests should render a user card when passed a user object 1`] = `
<div>
  <div
    class="my-8 rounded-lg border border-gray-200 bg-white p-8 shadow"
  >
    <ul
      class="flex flex-col space-y-2 text-slate-800"
    >
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.982 18.725A7.488 7.488 0 0 0 12 15.75a7.488 7.488 0 0 0-5.982 2.975m11.963 0a9 9 0 1 0-11.963 0m11.963 0A8.966 8.966 0 0 1 12 21a8.966 8.966 0 0 1-5.982-2.275M15 9.75a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="text-lg font-semibold"
        >
          Leanne Graham
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.75 6.75v10.5a2.25 2.25 0 0 1-2.25 2.25h-15a2.25 2.25 0 0 1-2.25-2.25V6.75m19.5 0A2.25 2.25 0 0 0 19.5 4.5h-15a2.25 2.25 0 0 0-2.25 2.25m19.5 0v.243a2.25 2.25 0 0 1-1.07 1.916l-7.5 4.615a2.25 2.25 0 0 1-2.36 0L3.32 8.91a2.25 2.25 0 0 1-1.07-1.916V6.75"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          data-testid="email"
        >
          Sincere@april.biz
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 10.5a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1 1 15 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          <span>
            Kulas Light
            , 
          </span>
          <span>
            Apt. 556
            , 
          </span>
          <span>
            Gwenborough
            , 
          </span>
          <span>
            92998-3874
          </span>
        </div>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 6.75c0 8.284 6.716 15 15 15h2.25a2.25 2.25 0 0 0 2.25-2.25v-1.372c0-.516-.351-.966-.852-1.091l-4.423-1.106c-.44-.11-.902.055-1.173.417l-.97 1.293c-.282.376-.769.542-1.21.38a12.035 12.035 0 0 1-7.143-7.143c-.162-.441.004-.928.38-1.21l1.293-.97c.363-.271.527-.734.417-1.173L6.963 3.102a1.125 1.125 0 0 0-1.091-.852H4.5A2.25 2.25 0 0 0 2.25 4.5v2.25Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span>
          1-770-736-8031 x56442
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 21a9.004 9.004 0 0 0 8.716-6.747M12 21a9.004 9.004 0 0 1-8.716-6.747M12 21c2.485 0 4.5-4.03 4.5-9S14.485 3 12 3m0 18c-2.485 0-4.5-4.03-4.5-9S9.515 3 12 3m0 0a8.997 8.997 0 0 1 7.843 4.582M12 3a8.997 8.997 0 0 0-7.843 4.582m15.686 0A11.953 11.953 0 0 1 12 10.5c-2.998 0-5.74-1.1-7.843-2.918m15.686 0A8.959 8.959 0 0 1 21 12c0 .778-.099 1.533-.284 2.253m0 0A17.919 17.919 0 0 1 12 16.5c-3.162 0-6.133-.815-8.716-2.247m0 0A9.015 9.015 0 0 1 3 12c0-1.605.42-3.113 1.157-4.418"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span>
          hildegard.org
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.25 14.15v4.25c0 1.094-.787 2.036-1.872 2.18-2.087.277-4.216.42-6.378.42s-4.291-.143-6.378-.42c-1.085-.144-1.872-1.086-1.872-2.18v-4.25m16.5 0a2.18 2.18 0 0 0 .75-1.661V8.706c0-1.081-.768-2.015-1.837-2.175a48.114 48.114 0 0 0-3.413-.387m4.5 8.006c-.194.165-.42.295-.673.38A23.978 23.978 0 0 1 12 15.75c-2.648 0-5.195-.429-7.577-1.22a2.016 2.016 0 0 1-.673-.38m0 0A2.18 2.18 0 0 1 3 12.489V8.706c0-1.081.768-2.015 1.837-2.175a48.111 48.111 0 0 1 3.413-.387m7.5 0V5.25A2.25 2.25 0 0 0 13.5 3h-3a2.25 2.25 0 0 0-2.25 2.25v.894m7.5 0a48.667 48.667 0 0 0-7.5 0M12 12.75h.008v.008H12v-.008Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p>
          Romaguera-Crona
        </p>
        <p>
          Multi-layered client-server neural-net
        </p>
      </li>
    </ul>
  </div>
</div>
`;

exports[`UserCard unit tests should render another user card correctly when passed another user object 1`] = `
<div>
  <div
    class="my-8 rounded-lg border border-gray-200 bg-white p-8 shadow"
  >
    <ul
      class="flex flex-col space-y-2 text-slate-800"
    >
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.982 18.725A7.488 7.488 0 0 0 12 15.75a7.488 7.488 0 0 0-5.982 2.975m11.963 0a9 9 0 1 0-11.963 0m11.963 0A8.966 8.966 0 0 1 12 21a8.966 8.966 0 0 1-5.982-2.275M15 9.75a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="text-lg font-semibold"
        >
          Glenna Reichert
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.75 6.75v10.5a2.25 2.25 0 0 1-2.25 2.25h-15a2.25 2.25 0 0 1-2.25-2.25V6.75m19.5 0A2.25 2.25 0 0 0 19.5 4.5h-15a2.25 2.25 0 0 0-2.25 2.25m19.5 0v.243a2.25 2.25 0 0 1-1.07 1.916l-7.5 4.615a2.25 2.25 0 0 1-2.36 0L3.32 8.91a2.25 2.25 0 0 1-1.07-1.916V6.75"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          data-testid="email"
        >
          Chaim_McDermott@dana.io
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 10.5a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M19.5 10.5c0 7.142-7.5 11.25-7.5 11.25S4.5 17.642 4.5 10.5a7.5 7.5 0 1 1 15 0Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          <span>
            Dayna Park
            , 
          </span>
          <span>
            Suite 449
            , 
          </span>
          <span>
            Bartholomebury
            , 
          </span>
          <span>
            76495-3109
          </span>
        </div>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 6.75c0 8.284 6.716 15 15 15h2.25a2.25 2.25 0 0 0 2.25-2.25v-1.372c0-.516-.351-.966-.852-1.091l-4.423-1.106c-.44-.11-.902.055-1.173.417l-.97 1.293c-.282.376-.769.542-1.21.38a12.035 12.035 0 0 1-7.143-7.143c-.162-.441.004-.928.38-1.21l1.293-.97c.363-.271.527-.734.417-1.173L6.963 3.102a1.125 1.125 0 0 0-1.091-.852H4.5A2.25 2.25 0 0 0 2.25 4.5v2.25Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span>
          (775)976-6794 x41206
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 21a9.004 9.004 0 0 0 8.716-6.747M12 21a9.004 9.004 0 0 1-8.716-6.747M12 21c2.485 0 4.5-4.03 4.5-9S14.485 3 12 3m0 18c-2.485 0-4.5-4.03-4.5-9S9.515 3 12 3m0 0a8.997 8.997 0 0 1 7.843 4.582M12 3a8.997 8.997 0 0 0-7.843 4.582m15.686 0A11.953 11.953 0 0 1 12 10.5c-2.998 0-5.74-1.1-7.843-2.918m15.686 0A8.959 8.959 0 0 1 21 12c0 .778-.099 1.533-.284 2.253m0 0A17.919 17.919 0 0 1 12 16.5c-3.162 0-6.133-.815-8.716-2.247m0 0A9.015 9.015 0 0 1 3 12c0-1.605.42-3.113 1.157-4.418"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span>
          conrad.com
        </span>
      </li>
      <li
        class="flex items-center space-x-2"
      >
        <svg
          aria-hidden="true"
          class="size-5 text-gray-700"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.25 14.15v4.25c0 1.094-.787 2.036-1.872 2.18-2.087.277-4.216.42-6.378.42s-4.291-.143-6.378-.42c-1.085-.144-1.872-1.086-1.872-2.18v-4.25m16.5 0a2.18 2.18 0 0 0 .75-1.661V8.706c0-1.081-.768-2.015-1.837-2.175a48.114 48.114 0 0 0-3.413-.387m4.5 8.006c-.194.165-.42.295-.673.38A23.978 23.978 0 0 1 12 15.75c-2.648 0-5.195-.429-7.577-1.22a2.016 2.016 0 0 1-.673-.38m0 0A2.18 2.18 0 0 1 3 12.489V8.706c0-1.081.768-2.015 1.837-2.175a48.111 48.111 0 0 1 3.413-.387m7.5 0V5.25A2.25 2.25 0 0 0 13.5 3h-3a2.25 2.25 0 0 0-2.25 2.25v.894m7.5 0a48.667 48.667 0 0 0-7.5 0M12 12.75h.008v.008H12v-.008Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p>
          Yost and Sons
        </p>
        <p>
          Switchable contextually-based project
        </p>
      </li>
    </ul>
  </div>
</div>
`;
